File: challenges/2026-08-21-longest-balanced-parity-subarray/solution.py
"""
Longest Balanced Parity Subarray

See problem.md for the full statement.
"""

from typing import List


def longest_balanced_parity_subarray(nums: List[int], k: int) -> int:
    """
    Return the length of the longest contiguous, non-empty subarray of
    `nums` whose balance (|#even - #odd|) is at most `k`.
    """
    # TODO: implement
    n = len(nums)
    even = 0
    odd = 0
    diff = 0
    return diff  # placeholder, remove me


def _run_self_checks():
    cases = [
        (([1, 2, 3, 4, 5], 1), 5),
        (([2, 4, 6, 1, 3], 0), 4),
        (([2, 4, 6, 8], 2), 2),
        # extra edge cases
        (([7], 5), 1),
        (([2, 4, 6, 8], 0), 0),
    ]

    all_passed = True
    for i, (args, expected) in enumerate(cases, 1):
        nums, k = args
        actual = longest_balanced_parity_subarray(list(nums), k)
        status = "PASS" if actual == expected else "FAIL"
        if status == "FAIL":
            all_passed = False
        print(f"Case {i}: {status} (nums={nums}, k={k} -> got {actual}, expected {expected})")

    if all_passed:
        print("All cases passed.")
    else:
        print("Some cases failed.")


if __name__ == "__main__":
    _run_self_checks()

File: challenges/2026-08-21-longest-balanced-parity-subarray/solution.ts
/**
 * Longest Balanced Parity Subarray
 *
 * See problem.md for the full statement.
 */

function longestBalancedParitySubarray(nums: number[], k: number): number {
  // TODO: implement
  const n = nums.length;
  let even = 0;
  let odd = 0;
  let diff = 0;
  return diff; // placeholder, remove me
}

function runSelfChecks(): void {
  const cases: { nums: number[]; k: number; expected: number }[] = [
    { nums: [1, 2, 3, 4, 5], k: 1, expected: 5 },
    { nums: [2, 4, 6, 1, 3], k: 0, expected: 4 },
    { nums: [2, 4, 6, 8], k: 2, expected: 2 },
    { nums: [7], k: 5, expected: 1 },
    { nums: [2, 4, 6, 8], k: 0, expected: 0 },
  ];

  let allPassed = true;
  cases.forEach((c, i) => {
    const actual = longestBalancedParitySubarray([...c.nums], c.k);
    const status = actual === c.expected ? "PASS" : "FAIL";
    if (status === "FAIL") allPassed = false;
    console.log(
      `Case ${i + 1}: ${status} (nums=${JSON.stringify(c.nums)}, k=${c.k} -> got ${actual}, expected ${c.expected})`
    );
  });

  console.log(allPassed ? "All cases passed." : "Some cases failed.");
}

runSelfChecks();
